lib/tapioca/dsl/compilers/graphql_input_object.rb

```diff
@@ -45,7 +45,13 @@ class GraphqlInputObject < Compiler
 
         sig { override.void }
         def decorate
-          arguments = constant.all_argument_definitions
+          graphql_gem = T.must(Gemfile.new([]).gem('graphql'))
+
+          # Skip methods explicitly defined in code
+          arguments = constant.all_argument_definitions.select do |argument|
+            source_location = constant.instance_method(argument.keyword.to_s).source_location&.first
+            source_location && graphql_gem.contains_path?(source_location)
+          end
```

Comment:
Open to suggestions on a better way to do this. Alternatives considered:

1. Not relying on the internal `Gemfile` class and just using bundler, something like:
```ruby
full_gem_path = Bundler.rubygems.find_name('graphql').first.full_gem_path
source_location && source_location.start_with?(full_gem_path)
```

2. Using `Gemfile::GemSpec.spec_lookup_by_file_path`, like:
```ruby
source_location && Gemfile::GemSpec.spec_lookup_by_file_path[source_location]&.name == 'graphql'
```

Reply:
I'm not familiar with `graphql`, how would it behave with subclassing?

Something like this:

```rb
class CreateCommentInput < GraphQL::Schema::InputObject
  argument :body, String, required: true
end

class CustomCreateCommentInput < CreateCommentInput
  def body; end
end
```

Could it mess up with the order of source locations?

Reply:
We basically only care about typing the methods if they come from the graphql source itself (specifically this block of code: https://github.com/rmosolgo/graphql-ruby/blob/89854fa911d8243bb216910c370dbed780c8c178/lib/graphql/schema/input_object.rb#L133-L139). If the method is defined at all in user-land we don't want to define it in RBI. In your example, if `def body` were:
- not defined at all in user-land, it should show up in RBI
- defined in `CreateCommentInput`, it should **not** show up in RBI
- defined in `CustomCreateCommentInput`, it show **not** show up in RBI

so I don't think the subclassing would have a negative effect.

Reply:
Looks great, thanks for cleaning that up!